builder/scripts/build-wheels

```diff
@@ -98,44 +96,79 @@ rm -rf output
 mkdir output
 cd output
 
-log "Processing package: ${PACKAGE}"
+# Function to save build sequence summary
+save_build_sequence_summary() {
+    local package="$1"
+    local summary_file="work-dir/build-sequence-summary.json"
+    
+    if [ -f "$summary_file" ]; then
+        # Create a package-specific filename
+        local safe_package_name=$(echo "$package" | sed 's/[^a-zA-Z0-9._-]/_/g')
+        local saved_summary="build-sequence-summary-${safe_package_name}.json"
+        log "Saving build sequence summary for $package to $saved_summary"
+        cp "$summary_file" "$saved_summary"
+        BUILD_SEQUENCE_SUMMARIES+=("$saved_summary")
+    fi
+}
 
-log "Resolving dependencies and creating build order"
-# Create bootstrap to resolve dependencies
-fromager_args=(
-    bootstrap
-    "${PACKAGE}"
-)
+# Process each package
+for PACKAGE in "${PACKAGES[@]}"; do
+    log "Processing package: ${PACKAGE}"
 
-# Add wheel server URL if provided
-if [ -n "$WHEEL_SERVER_URL" ]; then
-    fromager_args+=(--cache-wheel-server-url "$WHEEL_SERVER_URL")
-fi
+    log "Resolving dependencies and creating build order"
+    # Create bootstrap to resolve dependencies
+    fromager_args=(
+        bootstrap
+        "${PACKAGE}"
+    )
 
-echo "Running: fromager ${fromager_args[*]}"
-fromager "${fromager_args[@]}"
+    # Add wheel server URL if provided
+    if [ -n "$WHEEL_SERVER_URL" ]; then
+        fromager_args+=(--cache-wheel-server-url "$WHEEL_SERVER_URL")
+    fi
 
-log "Building wheels from build order"
+    echo "Running: fromager ${fromager_args[*]}"
+    fromager "${fromager_args[@]}"
 
-# Build wheels using the generated build order
-fromager_build_args=(
-    build-sequence
-    work-dir/build-order.json
-)
+    log "Building wheels from build order"
 
-if [ -n "$WHEEL_SERVER_URL" ]; then
-    fromager_build_args+=(--cache-wheel-server-url "$WHEEL_SERVER_URL")
-fi
+    # Build wheels using the generated build order
+    fromager_build_args=(
+        build-sequence
+        work-dir/build-order.json
+    )
 
-echo "Running: fromager ${fromager_build_args[*]}"
-fromager "${fromager_build_args[@]}"
+    if [ -n "$WHEEL_SERVER_URL" ]; then
+        fromager_build_args+=(--cache-wheel-server-url "$WHEEL_SERVER_URL")
+    fi
+
+    echo "Running: fromager ${fromager_build_args[*]}"
+    fromager "${fromager_build_args[@]}"
+
+    # Save build sequence summary before clearing work-dir
+    save_build_sequence_summary "$PACKAGE"
+
+    # Clear work-dir for next package (but preserve sdists-repo and wheels-repo)
+    if [ "${#PACKAGES[@]}" -gt 1 ] && [ "$PACKAGE" != "${PACKAGES[-1]}" ]; then
```

Comment:
I don't think we need to do this check. At the end of this loop, just remove the `work-dir` if it's there.